sdl_android/src/main/java/com/smartdevicelink/managers/SdlManager.java

```diff
@@ -129,94 +127,137 @@ public void onError(String info, Exception e){
+				transitionToState(BaseSubManager.LIMITED);
+				notifyDevListener(null);
+			}
+		} else {
+			// We should never be here, but somehow one of the sub-sub managers is null
+			String info = "ERROR one of the sdl sub managers is null";
+			DebugTool.logError(info);
+			transitionToState(BaseSubManager.ERROR);
+			notifyDevListener(info);
+		}
+	}
+
+	private void notifyDevListener(String info) {
+		if (managerListener != null) {
+			if (state == BaseSubManager.ERROR){
+				managerListener.onError(info, null);
+			} else {
+				managerListener.onStart();
+			}
+		}
+
+		// Set the app icon
+		if (state == BaseSubManager.READY || state == BaseSubManager.LIMITED) {
+			setAppIcon();
+		}
+	}
+
+	private void setAppIcon(){
+		if (SdlManager.this.appIcon != null && SdlManager.this.appIcon.getName() != null) {
+			if (!fileManager.hasUploadedFile(SdlManager.this.appIcon)) {
+				fileManager.uploadArtwork(SdlManager.this.appIcon, new CompletionListener() {
+					@Override
+					public void onComplete(boolean success) {
+						if (success) {
+							SetAppIcon msg = new SetAppIcon(SdlManager.this.appIcon.getName());
+							_internalInterface.sendRPCRequest(msg);
+						}
 					}
-				}
+				});
+			} else {
+				SetAppIcon msg = new SetAppIcon(SdlManager.this.appIcon.getName());
+				_internalInterface.sendRPCRequest(msg);
 			}
 		}
-	};
+	}
 
 	protected void initialize(){
-		// instantiate managers
-
+		// Instantiate sub managers
 		this.permissionManager = new PermissionManager(_internalInterface);
-		this.permissionManager.start(subManagerListener);
-
 		this.fileManager = new FileManager(_internalInterface, context);
-		this.fileManager.start(subManagerListener);
-
 		if (lockScreenConfig.isEnabled()) {
 			this.lockScreenManager = new LockScreenManager(lockScreenConfig, context, _internalInterface);
-			this.lockScreenManager.start(subManagerListener);
 		}
-
 		this.screenManager = new ScreenManager(_internalInterface, this.fileManager);
-		this.screenManager.start(subManagerListener);
-
-
 		if(getAppTypes().contains(AppHMIType.NAVIGATION) || getAppTypes().contains(AppHMIType.PROJECTION)){
 			this.videoStreamingManager = new VideoStreamingManager(_internalInterface);
-		}else{
+		} else {
 			this.videoStreamingManager = null;
 		}
-
-
 		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN
 				&& (getAppTypes().contains(AppHMIType.NAVIGATION) || getAppTypes().contains(AppHMIType.PROJECTION)) ) {
 			this.audioStreamManager = new AudioStreamManager(_internalInterface, context);
 		} else {
 			this.audioStreamManager = null;
 		}
 
+		// Start sub managers
+		this.permissionManager.start(subManagerListener);
+		this.fileManager.start(subManagerListener);
+		if (lockScreenConfig.isEnabled()){
+			this.lockScreenManager.start(subManagerListener);
+		}
+		this.screenManager.start(subManagerListener);
+	}
+
+	public int getState() {
```

Comment:
Can we add some javadoc to this public method? Probably with `#see` tags to the states